agent/http.go

```diff
@@ -41,6 +42,18 @@ type HTTPServer struct {
 	proto string
 }
 
+type RedirectFS struct {
+	fs http.FileSystem
+}
+
+func (fs *RedirectFS) Open(name string) (http.File, error) {
+	file, err := fs.fs.Open(name)
+	if err != nil {
+		file, err = fs.fs.Open("/index.html")
```

Comment:
Was this copied from another project? I know we do similar things in other places for example https://github.com/hashicorp/terraform-registry/blob/master/helper/fallback-fs/fallback.go.

Reply:
Nope, I just implemented it myself. I didn't think to look for this particular thing in another project.